### ISSUE TITLE
Multiple Event Consumer

### ISSUE THREAD

**Issue by amitranjan3** [resilience4j/resilience4j#1646]
Resilience4j version:

Java version: 1.8
<img width="1491" alt="Screenshot 2022-03-18 at 4 29 23 AM" src="https://user-images.githubusercontent.com/99726415/158908541-caa1e289-8b2a-438c-b771-39f412d77dd8.png">

<img width="1187" alt="Screenshot 2022-03-18 at 4 29 41 AM" src="https://user-images.githubusercontent.com/99726415/158908254-7a0945dd-a640-405a-b0b6-218d4783292d.png">

Hi 
i am trying to log the onstate transition like closed to open and others but instead of printing 1 time it is printing multiple time (same message) 

i tried many way to resolve this , is it dafault feature or i am missing something ?

Thanks & Regards

**Comment by RobWin** [comment]
Hi,

duplicate of https://github.com/resilience4j/resilience4j/issues/1102

**Comment by RobWin** [comment]
Could you check if this code block is invoked multiple times by Spring?

**Comment by amitranjan3** [comment]
Hi @RobWin 

I have added this code in @configuration, could you please help me how to resolve this?
<img width="1203" alt="Screenshot 2022-03-18 at 11 44 47 PM" src="https://user-images.githubusercontent.com/99726415/159060130-9d0d635a-75a1-4bf7-9543-8a4185025b8c.png">

**Comment by amitranjan3** [comment]
Hi @RobWin any suggestions

**Comment by RobWin** [comment]
My suggestion is the comment above. Did you do it?

**Comment by amitranjan3** [comment]
Hi @RobWin, i have logged in return as you said in above comment , this code block is invoked only once when you start you spring server , 
<img width="1152" alt="Screenshot 2022-03-20 at 6 24 43 PM" src="https://user-images.githubusercontent.com/99726415/159164058-16f41753-972c-4b40-87bc-2f8b6774eb4b.png">

i have changed minimumNumberOfCalls from 3 to 5 . so initially i was giving logs 3 times and now you can see logs is 5 times, so basically it is storing all the event before any transition happens
<img width="964" alt="Screenshot 2022-03-20 at 6 39 30 PM" src="https://user-images.githubusercontent.com/99726415/159163832-6d184479-c356-4148-ae3e-0344c6ecd634.png">

**Comment by amitranjan3** [comment]
@RobWin 
and this code is written in my serviceimplention 
 return circuitBreakerFactory.create("TEST").run(reactiveHttpService.postAndExchange(module.getBaseUrl(), module.getApiUri(), requestBody, null), t -> {
			return Mono.just(ClientResponse.create(HttpStatus.BAD_GATEWAY).build());
		});

**Comment by amitranjan3** [comment]
Hi @RobWin in the case you missed this thread. please let me know if i am missing something

**Comment by RobWin** [comment]
Thank you.
Could you add a last log statement in front of `circuitBreaker -> circuitBreaker.getEventPublisher().onStateTransition`

You are using Spring Cloud Circuitbreaker. The project is not created and maintained by us.
